src/lib.rs

```diff
@@ -78,6 +92,16 @@ impl UnixStream {
             }
         }
     }
+
+    /// Create an unnamed pair of connected sockets.
+    ///
+    /// Returns two `UnixStream`s which are connected to each other.
+    pub fn unnamed() -> io::Result<[UnixStream; 2]> {
```

Comment:
I think in this case the fact that it's homogeneous is somewhat relevant. For example you might iterate over the sockets to set some socket options. The important thing to me is that you can destructure it in an irrefutable pattern, which is now the case (although it wasn't for a long time).